index.d.ts

```diff
@@ -0,0 +1,579 @@
+import * as router from "router"
+
+// tslint:disable-next-line
+declare namespace hyperdom {
+
+  export interface VdomFragment {
+    children: any[]
+  }
+
+  abstract class HyperdomComponent {
+    public refreshImmediately (): void
+    public refreshComponent (): void
+    public refresh (): void
+  }
+
+  export abstract class RenderComponent extends HyperdomComponent {
+    public abstract render (): VdomFragment | Component
+  }
+
+  export abstract class RoutesComponent extends HyperdomComponent {
+    public abstract routes (): router.Route[]
+
+    protected renderLayout (content: any): VdomFragment | Component
+  }
+
+  export type Component = RoutesComponent | RenderComponent
+
+  interface DomAttachement {
+    remove (): void
+
+    detach (): void
+  }
+
+  export interface MountOpts {
+    requestRender?: (render: () => void) => void
+    window?: Window
+    document?: Document
+    router?: typeof router
+  }
+
+  export interface ObjectBinding {
+    get (): any
+
+    set (value: any): void
+  }
+
+  export type SimpleBinding = [object, string] | [object, string, (param: string | number) => any]
```

Comment:
`(param: string | number) => any` let's check if all input elements have value types of either string or number